core/lib/server/form.ts

```diff
@@ -26,10 +26,15 @@ import { _getPubFields } from "./pubFields";
 import { getUser } from "./user";
 
 /**
- * Get a form by either slug or id
+ * Get a form by either slug, id, or pubtype ID. If given a pubtype ID, retrieves the
+ * default form for that pubtype.
  */
 export const getForm = (
-	props: XOR<{ slug: string }, { id: FormsId }> & { communityId: CommunitiesId },
+	props: (
+		| { id: FormsId; slug?: never; pubTypeId?: never }
+		| { id?: never; slug: string; pubTypeId?: never }
+		| { id?: never; slug?: never; pubTypeId: PubTypesId }
```

Comment:
Do you know the difference between this approach the following?

```ts
| { id: FormsId }
| { slug: string }
| { pubTypeId: PubTypesId }
```

Is it better intellisense? The former seems much more concise.

Reply:
yup better intellisense, otherwise if you do a `props.slug` you get errors like

```
Property 'slug' does not exist on type '({ id: FormsId; } | { slug: string; } | { pubTypeId: PubTypesId; }) & { communityId: CommunitiesId; }'.
  Property 'slug' does not exist on type '{ id: FormsId; } & { communityId: CommunitiesId; }'
```